crates/server/src/rpc/mod.rs

```diff
@@ -45,30 +42,14 @@ pub async fn run_rpc_server(
         config,
         db,
         redis_events_queue,
-        quest_subscriptions: Arc::new(RwLock::new(HashMap::new())),
-        subscription_by_user_address: Arc::new(RwLock::new(HashMap::new())),
+        redis_channel_subscriber,
+        subscriptions_handle_by_transport_id: Arc::new(RwLock::new(HashMap::new())),
+        subscription_by_transport_id: Arc::new(RwLock::new(HashMap::new())),
```

Comment:
If both fields are maps with the same keys, I think it would be better to merge them into another structure.
```suggestion
       struct TransportContext {
          subscription: ...
          subscription_handle: ...
       }
       ... 
        transport_contexts: ...
```